specifier.go

```diff
@@ -91,6 +91,7 @@ func newSpecifiers(v string, santizer func(string) string, opts ...SpecifierOpti
 		if strings.TrimSpace(vv) == "*" {
 			vv = ">=0.0.0"
 		}
+		vv = strings.ReplaceAll(vv, "-", ".")
 
 		// Validate the segment
 		if !validConstraintRegexp.MatchString(vv) {
```

Comment:
Is it possible to run the `sanitizer` before this valid constraint check? Agreed with Jon that this is very CRAN/R specific, yeah. The sanitizers already have the logic to normalize specifiers specific to each ecosystem though, so I just thought the order could be flipped.

Reply:
Looking more, it's not super straightforward because the sanitizer works on versions only and not full specifiers, so you'd have to parse the version out of the specifier first, and rebuild the specifier with the sanitized version. `newSpecifier()` also runs the sanitizer so that would be redundant. A few things would probably need to be moved around..

 But reusing the sanitizer rather than duplicating it here for a special case does feel nicer to me. `SanitizeCranVersion` is also doing a few more things besides just replacing dashes with periods.

Reply:
Alternatively, I wonder what would happen if we loosened up the valid constraint regex. It validates both the operator and the version, but what if it validated the operator only? Later on, the version gets parsed and presumably validated  through the version parser, https://github.com/rstudio/go-pep440-version/blob/39e6833d241fbf3879538c68fa3b8a364bb74fc1/specifier.go#L135-L153

Even the operator gets validated via `validate()`. I'm not exactly sure why the initial regex check is needed when it seems redundant.